logger/logger.go

```diff
@@ -173,6 +173,7 @@ func write(stdout bool, msg string, args ...interface{}) {
 		if machineReadable {
 			strs := strings.Split(fmt.Sprintf(msg, args...), "\n")
 			for _, m := range strs {
+				m = escapeInvalidJSONChars(m)
 				fmt.Printf("{\"type\": \"out\", \"message\": \"%s\"}\n", strings.Trim(m, "\n "))
```

Comment:
Instead of creating a json object ourselves, if we use Golang's Json encoder and decoder, it may automatically escape the special characters.

Reply:
Please look at the latest changes.